public/js/save.js

```diff
@@ -124,26 +170,48 @@ $(document).ready(function() {
         counter++;
     }
     //Push All Data
-    function pushData() { //TODO: Check for connectivity before running.
+    function pushData(alertUser) {
         var error = false;
-        for (var i = (counter-1); i >= 1; i--) {
-            var data = JSON.parse(get('form', i));
-            var request = $.ajax({
-                type: 'POST',
-                url: '/pushData',
-                data: data
-            });
-            request.done(function(response) { //If pushing is successful.
-                console.log("Data successfully pushed.");
-            });
-            request.fail(function(jqXHR, textStatus) { //If pushing is unsuccessful.
-                if(!error){
-                    alert("Error while pushing. Please reach out to a qualified individual for assistance. Error Message: " + textStatus);
-                    error = true;
-                }
-            });
+        var dataArray = [];
+        for (var i = 0; i < counter; i++) {
+            dataArray.push(get('form', i));
         }
-        counter = 1;
-        localStorage.clear();
+        dataArray = JSON.stringify(dataArray);
+        var request = $.ajax({
+            type: 'POST',
+            url: '/pushData',
+            data: {
+                content: dataArray
+            },
+        });
+        request.done(function(response) { //If pushing is successful.
+            if(response === "No Email"){
+                if(alertUser){
+                    alert("Please sign in with google.");
+                }else{
+                    console.log("Please sign in with google.");
+                }
+            }else if(response === "Invalid Email"){
+                if(alertUser){
+                    alert("Please sign in with your nuevaschool.org account");
+                }else{
+                    console.log("Please sign in with your nuevaschool.org account");
+                }
+            }else{
+                console.log(response);
```

Comment:
Maybe an alert like "unexpected response"? All expected outcomes should be covered by ifs and the final else should be the exception, the unexpected response.